src/Sulu/Bundle/ContactBundle/Resources/js/containers/ContactAccountSelection/ContactAccountSelection.js

```diff
@@ -0,0 +1,158 @@
+// @flow
+import React, {Fragment} from 'react';
+import {observer} from 'mobx-react';
+import {action, computed, observable, toJS} from 'mobx';
+import equals from 'fast-deep-equal';
+import {MultiItemSelection} from 'sulu-admin-bundle/components';
+import {MultiListOverlay} from 'sulu-admin-bundle/containers';
+import {translate} from 'sulu-admin-bundle/utils';
+import ContactAccountSelectionStore from './stores/ContactAccountSelectionStore';
+import contactAccountSelectionStyles from './contactAccountSelection.scss';
+
+type Props = {|
+    disabled: boolean,
+    onChange: (value: Array<Object>) => void,
+    value: Array<Object>,
+|};
+
+@observer
+class ContactAccountSelection extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+        value: [],
+    };
+
+    @observable openedOverlay: ?string = undefined;
+    store: ContactAccountSelectionStore;
+
+    constructor(props: Props) {
+        super(props);
+
+        const {value} = this.props;
+
+        this.store = new ContactAccountSelectionStore();
+        this.store.loadItems(value);
+    }
+
+    componentDidUpdate() {
+        const {value} = this.props;
+
+        const newIds = toJS(value);
+        const loadedIds = this.loadedIds;
+
+        newIds.sort();
+        loadedIds.sort();
+        if (!equals(newIds, loadedIds) && !this.store.loading) {
+            this.store.loadItems(value);
+        }
+    }
+
+    @computed get loadedIds(): Array<string> {
+        return toJS(this.store.items.map((item) => item.id));
+    }
+
+    @action handleAddButtonClick = (type: ?string) => {
+        this.openedOverlay = type;
+    };
+
+    @action handleOverlayClose = () => {
+        this.openedOverlay = undefined;
+    };
+
+    @action handleConfirm(items: Array<Object>, prefix: string) {
+        const {onChange, value} = this.props;
+
+        const itemIds = items.map((contact) => prefix + contact.id);
+
+        onChange([
+            ...value.filter((id) => !id.startsWith(prefix) || itemIds.includes(id)),
+            ...itemIds,
+        ].filter((item, index, items) => index == items.indexOf(item)));
```

Comment:
wouldnt it be easier to just use `...itemIds.filter((id) => !value.includes(id))` above?

Reply:
I felt that it is easier to understand if I do that in two separate steps... Apart from that I am not totally sure if that would work :thinking:

Reply:
Tested it, seems to work, will change it.